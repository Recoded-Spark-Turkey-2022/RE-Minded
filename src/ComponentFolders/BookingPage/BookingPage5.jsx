import { React, useState } from 'react';
import { Link, useNavigate } from 'react-router-dom';
import { useTranslation } from 'react-i18next';
import Card1 from './Card';
// import ChoicesB from './QuestionChoicesB';

function BookingPage5() {
  const [checkboxes, setCheckboxes] = useState({
    option1: false,
    option2: false,
    option3: false,
    option4: false,
    option5: false,
    option6: false,
  });
  const navigate = useNavigate();

  function handleCheckboxChange(event) {
    const { name, checked } = event.target;
    setCheckboxes({
      ...checkboxes,
      [name]: checked,
    });
  }

  function handleClick() {
    if (Object.values(checkboxes).some((val) => val === true)) {
      navigate('/booking6');
    } else {
      alert('Please select at least one!');
    }
  }
  
  const { t } = useTranslation();

  return (
    <div className="flex flex-col items-center p-12">
      <Card1
        header={t('booking.header')}
        content={t('booking.content')}
      />
      <div className="font-poppins shadow-zinc-300 flex flex-col w-full md:max-w-2xl lg:max-w-4xl my-16 mx-auto px-8 py-4 shadow-md">
        <h1 className="text-xl pb-6 md:text-2xl lg:text-3xl lg:flex justify-center mt-4">
        {t('booking5.h1')}
        </h1>
        <form className="flex flex-col pl-6 text-md md:text-lg  lg:text-xl">
          <label htmlFor="one">
            <input
              type="checkbox"
              className="m-2 mb-6 mt-12"
              name="option1"
              checked={checkboxes.option1}
              onChange={handleCheckboxChange}
            />
            {t('booking5.b1')}
          </label>
          <label htmlFor="two">
            <input
              type="checkbox"
              className="m-2 mb-6"
              name="option2"
              checked={checkboxes.option2}
              onChange={handleCheckboxChange}
            />
            {t('booking5.b2')}
          </label>
          <label htmlFor="three">
            <input
              type="checkbox"
              className="m-2 mb-6"
              name="option3"
              checked={checkboxes.option3}
              onChange={handleCheckboxChange}
            />
            {t('booking5.b3')}
          </label>
          <label htmlFor="four">
            <input
              type="checkbox"
              className="m-2 mb-6"
              name="option4"
              checked={checkboxes.option4}
              onChange={handleCheckboxChange}
            />
            {t('booking5.b4')}
          </label>
          <label htmlFor="five">
            <input
              type="checkbox"
              className="m-2 mb-6"
              name="option5"
              checked={checkboxes.option5}
              onChange={handleCheckboxChange}
            />
            {t('booking5.b5')}
          </label>
          <label htmlFor="six">
            <input
              type="checkbox"
              className="m-2 mb-6"
              name="option6"
              checked={checkboxes.option6}
              onChange={handleCheckboxChange}
            />
            {t('booking5.b6')}
          </label>
        </form>
        <div className="flex justify-center md:justify-between lg:justify-between ">
          <Link to="/booking4">
            <button
              type="button"
              className="m-6 rounded-md box-border text-xl pr-6 pl-6 pt-2 pb-2 transition-all duration-250 bg-cyan-400 hover:bg-cyan-500 lg:text-2xl"
            >
              {t('booking.previous')}
            </button>
          </Link>
          <button
            type="button"
            onClick={handleClick}
            className="m-6 rounded-md box-border text-xl pr-6 pl-6 pt-2 pb-2 transition-all duration-250 bg-cyan-400 hover:bg-cyan-500 lg:text-2xl"
          >
            {t('booking.next')}
          </button>
        </div>
      </div>
    </div>
  );
}

export default BookingPage5;
