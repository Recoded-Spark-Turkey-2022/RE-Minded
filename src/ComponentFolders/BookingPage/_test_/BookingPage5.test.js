import React from 'react';
import renderer from 'react-test-renderer';
import BookingPage5 from '../BookingPage5';

test('Booking5 Page snapshot test', () => {
  const component = renderer.create(<BookingPage5 />);

  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
