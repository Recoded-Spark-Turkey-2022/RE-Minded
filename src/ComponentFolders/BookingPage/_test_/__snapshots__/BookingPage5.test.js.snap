// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Booking1 Page snapshot test 1`] = `
<div
  className="flex flex-col items-center p-12"
>
  <div
    className=" flex flex-col p-14 font-poppins "
  >
    <h1
      className="text-2xl pb-6 md:text-2xl lg:text-4xl"
    >
      LET'S MATCH YOU WITH THE RIGHT THERAPIST
    </h1>
    <p
      className="text-lg  md:text-xl opacity-50 lg:text-2xl lg:pr-40  "
    >
      Please fill out this short questionnaire to provide some general and anonymous background about you and the issues you'd like to deal with in online therapy. It would help us match you with the most suitable therapist for you. 
    </p>
  </div>
  <div
    className="font-poppins shadow-zinc-300 flex flex-col w-full md:max-w-2xl lg:max-w-4xl my-16 mx-auto px-8 py-4 shadow-md"
  >
    <div
      className="p-10"
    >
      <h1
        className="text-xl pb-6 md:text-2xl lg:text-3xl lg:flex justify-center mt-4"
      >
        What type of counseling are you looking for?
      </h1>
      <div
        className="flex flex-col pl-6 text-md md:text-lg lg:text-xl"
      >
        <div
          className="shadow-zinc-300 flex flex-col w-full mb-6 mt-12 px-4 py-4 shadow-md hover:bg-cyan-400 cursor-pointer"
        >
          Individual counseling
        </div>
        <div
          className="shadow-zinc-300 flex flex-col w-full mb-6 px-4 py-4 shadow-md hover:bg-cyan-400 cursor-pointer"
        >
          Teen counseling (for my child)
        </div>
      </div>
    </div>
    <div
      className="flex justify-center md:justify-start lg:justify-start "
    >
      <button
        className="m-6 rounded-md box-border text-xl pr-6 pl-6 pt-2 pb-2 transition-all duration-250 bg-cyan-400 hover:bg-cyan-500 lg:text-2xl"
        type="button"
      >
        NEXT
      </button>
    </div>
  </div>
</div>
`;

exports[`Booking5 Page snapshot test 1`] = `
<div
  className="flex flex-col items-center p-12"
>
  <div
    className=" flex flex-col p-14 font-poppins "
  >
    <h1
      className="text-2xl pb-6 md:text-2xl lg:text-4xl"
    >
      LET'S MATCH YOU WITH THE RIGHT THERAPIST
    </h1>
    <p
      className="text-lg  md:text-xl opacity-50 lg:text-2xl lg:pr-40  "
    >
      Please fill out this short questionnaire to provide some general and anonymous background about you and the issues you'd like to deal with in online therapy. It would help us match you with the most suitable therapist for you. 
    </p>
  </div>
  <div
    className="font-poppins shadow-zinc-300 flex flex-col w-full md:max-w-2xl lg:max-w-4xl my-16 mx-auto px-8 py-4 shadow-md"
  >
    <div
      className="p-10"
    >
      <h1
        className="text-xl pb-6 md:text-2xl lg:text-3xl lg:flex justify-center mt-4 "
      >
        Are there any issues you'd like to focus on?
      </h1>
      <div
        className="flex flex-col pl-6 text-md md:text-lg  lg:text-xl"
      >
        <label
          htmlFor="one"
        >
          <input
            className="m-2 mb-6 mt-12"
            type="radio"
          />
          Depression
        </label>
        <label
          htmlFor="two"
        >
          <input
            className="m-2 mb-6"
            type="radio"
          />
          Stress and Anxiety
        </label>
        <label
          htmlFor="three"
        >
          <input
            className="m-2 mb-6"
            type="radio"
          />
          Relationship issues
        </label>
        <label
          htmlFor="four"
        >
          <input
            className="m-2 mb-6"
            type="radio"
          />
          Family conflicts
        </label>
        <label
          htmlFor="five"
        >
          <input
            className="m-2 mb-6"
            type="radio"
          />
          Trauma and abuse
        </label>
        <label
          htmlFor="six"
        >
          <input
            className="m-2 mb-4"
            type="radio"
          />
          Eating disorders
        </label>
      </div>
    </div>
    <div
      className="flex justify-center md:justify-start lg:justify-start "
    >
      <button
        className="m-6 rounded-md box-border text-xl pr-6 pl-6 pt-2 pb-2 transition-all duration-250 bg-cyan-400 hover:bg-cyan-500 lg:text-2xl"
        type="button"
      >
        NEXT
      </button>
    </div>
  </div>
</div>
`;
